third_party/nvfuser/csrc/scheduler/registry.cpp

```diff
@@ -34,24 +34,52 @@ std::deque<std::deque<TensorView*>> tvChains(
   return tv_chains;
 }
 
-bool rejectScheduleForTorchGather(
-    TorchGatherOp* torch_gather,
+// bool rejectScheduleForTorchGather(
+template<typename Op>
+bool rejectScheduleFusionInputRequirement(
+    Op* op,
```

Comment:
Does `Op` need to be a template type?

Reply:
Ha this is a good catch!

I guess we are only using `getOpString` and `input` on these and I should have just used an Expr*.